pkg/cmd/server/origin/health_grpc.go

```diff
@@ -0,0 +1,161 @@
+package origin
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	goruntime "runtime"
+	"strings"
+	"time"
+
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+)
+
+func NewGRPCStuckThreads() *TooManyThreadsStuckHealth {
+	return &TooManyThreadsStuckHealth{
+		name:           "too-many-gprc-connections",
+		targetFunction: "google.golang.org/grpc/transport.(*http2Client).NewStream",
+		limit:          5000,
```

Comment:
I know we saw well over 20k on 3.7, yesterday on 3.9 we saw 10k.  I'm wondering if we should set this to something higher, like 10k, so to make sure there are minimized false positives. Is 5k enough? I don't know.

Reply:
Given the risk (however small) of false positives, I suggest we add a toggle for the feature and a knob for the limit.

Reply:
> Given the risk (however small) of false positives, I suggest we add a toggle for the feature and a knob for the limit.

Sure.